docs/source/app-dev/authorization.rst

```diff
@@ -7,124 +7,195 @@ Authorization
+
+
+User access tokens
+==================
+
+Daml ledger that support participant user management also accept user access tokens.
+They are useful for scenarios where an application's rights change dynamically over the application's lifetime.
+
+User access tokens do not encode rights directly like custom Daml claims tokens.
+Instead, user access tokens encode the participant user on behalf of which the application is issuing a request.
+
+When handling such requests, participant nodes look up the participant user's current rights
+before checking request authorization per the  :ref:`table above <authorization-claims>`.
+Thus the rights granted to an application can be changed dynamically using
+the participant user management service *without* issuing new access tokens,
+as would be required for the custom Daml claims tokens explained below.
+
+User access tokens are `JWTs <https://datatracker.ietf.org/doc/html/rfc7519>`_ that follow the
+`OAuth 2.0 standard <https://datatracker.ietf.org/doc/html/rfc6749>`_ with a JSON payload of the following format.
+
+.. code-block:: json
+
+   {
+      "aud": "someParticipantId",
+      "sub": "someUserId",
+      "exp": 1300819380
+      "scope": "daml_ledger_api"
+   }
+
+where
+
+- ``aud`` is an optional field, which restricts the token to participant nodes with the given id
+- ``sub`` is a required field, which specifies the participant user's id
+- ``exp`` is an optional field, which specifies the JWT expiration date (in seconds since EPOCH)
+- ``scope`` is a space-separated list of `OAuth 2.0 scopes <https://datatracker.ietf.org/doc/html/rfc6749#section-3.3>`_
+  that must contain the ``"daml_ledger_api"`` scope
+
+
+Custom Daml claims access tokens
+================================
+
+This format represents the :ref:`rights <authorization-claims>` granted by the access token as custom claims in the JWT's payload, like so:
+
+
+.. code-block:: json
+
+   {
+      "https://daml.com/ledger-api": {
+        "ledgerId": null,
+        "participantId": "123e4567-e89b-12d3-a456-426614174000",
+        "applicationId": null,
+        "admin": true,
+        "actAs": ["Alice"],
+        "readAs": ["Bob"]
+      },
+      "exp": 1300819380
+   }
+
+where all of the fields are optional, and if present,
+
+- ``ledgerId`` and ``participantId`` restrict the validity of the token to the given ledger or participant node
+- ``applicationId`` requires requests with this token to use that application id or not set an application id at all, which should be used to distinguish requests from different applications
+- ``exp`` is the standard JWT expiration date (in seconds since EPOCH)
+- ``actAs``, ``readAs`` and (participant) ``admin`` encode the rights granted by this access token
+
+The ``public`` right is implicitly granted to any bearing a valid JWT issued by a trusted issuer (even without being an admin or being able to act or read on behalf of any party).
```

Comment:
```suggestion
The ``public`` right is implicitly granted to anyone bearing a valid JWT issued by a trusted issuer (even without being an admin or being able to act or read on behalf of any party).
```

Is this true? Don't we even check participantId or applicationId or anything? Could I use an empty token or a token issued for a completely different audience by the same token issuer?

Reply:
> Is this true? Don't we even check participantId or applicationId or anything? Could I use an empty token or a token issued for a completely different audience by the same token issuer?

We do check not expired, ledgerId, and participantId for a token to be considered valid: https://github.com/digital-asset/daml/blob/2deeabf3fac6ec6d75f9d779f4acde1c0dacc56b/ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/Authorizer.scala#L43

Moreover, every request that does specify an `applicationId` compares that one to the one in the token if set.

I'll clarify the text in a separate PR.

Reply:
https://github.com/digital-asset/daml/pull/12534